### ISSUE TITLE
max_line_length = 0 flattens a file to a single column

### ISSUE THREAD

**Issue by paulmfischer** [sindresorhus/atom-editorconfig#139]
I have the below editorconfig file and the *.md section has `max_line_length = 0` which technically isn't valid according to the [wiki](https://github.com/editorconfig/editorconfig/wiki/EditorConfig-Properties#max_line_length).  Should this plugin handle that case in a different way or ignore an invalid value?  Obviously I need to use `off` instead of `0` and have done that but thought I would open an issue here as well about this edge case.

![xavierreadme](https://cloud.githubusercontent.com/assets/6816413/20015651/41ce3dd4-a28a-11e6-8d83-536eabd9af5e.png)

### Involved .editorconfig-files

```
# Editor configuration, see http://editorconfig.org
root = true

[*]
charset = utf-8
indent_style = space
indent_size = 4
end_of_line = lf
insert_final_newline = true
trim_trailing_whitespace = true

[*.json]
indent_size = 2

[*.ts]
max_line_length = 140

[*.md]
max_line_length = 0
trim_trailing_whitespace = false
```

### Directory structure

```
Please replace this by the output of `tree -a .` or `tree . \f`
```

### Installed packages

```
Built-in Atom Packages (89)
├── atom-dark-syntax@0.27.0
├── atom-dark-ui@0.52.0
├── atom-light-syntax@0.28.0
├── atom-light-ui@0.44.0
├── base16-tomorrow-dark-theme@1.2.0
├── base16-tomorrow-light-theme@1.2.0
├── one-dark-ui@1.6.0
├── one-light-ui@1.6.0
├── one-dark-syntax@1.3.0
├── one-light-syntax@1.3.0
├── solarized-dark-syntax@1.0.2
├── solarized-light-syntax@1.0.2
├── about@1.7.0
├── archive-view@0.61.1
├── autocomplete-atom-api@0.10.0
├── autocomplete-css@0.11.2
├── autocomplete-html@0.7.2
├── autocomplete-plus@2.31.1
├── autocomplete-snippets@1.11.0
├── autoflow@0.27.0
├── autosave@0.23.1
├── background-tips@0.26.1
├── bookmarks@0.42.0
├── bracket-matcher@0.82.1
├── command-palette@0.38.0
├── deprecation-cop@0.54.1
├── dev-live-reload@0.47.0
├── encoding-selector@0.22.0
├── exception-reporting@0.40.0
├── find-and-replace@0.201.1
├── fuzzy-finder@1.4.0
├── git-diff@1.1.0
├── go-to-line@0.31.0
├── grammar-selector@0.48.2
├── image-view@0.59.0
├── incompatible-packages@0.26.1
├── keybinding-resolver@0.35.0
├── line-ending-selector@0.5.0
├── link@0.31.1
├── markdown-preview@0.158.0
├── metrics@1.0.0
├── notifications@0.65.1
├── open-on-github@1.2.0
├── package-generator@1.0.0
├── settings-view@0.242.2-hotfix1
├── snippets@1.0.2
├── spell-check@0.68.2
├── status-bar@1.4.1
├── styleguide@0.47.0
├── symbols-view@0.113.1
├── tabs@0.101.0
├── timecop@0.33.2
├── tree-view@0.209.3
├── update-package-dependencies@0.10.0
├── welcome@0.35.1
├── whitespace@0.33.0
├── wrap-guide@0.38.2
├── language-c@0.52.1
├── language-clojure@0.21.0
├── language-coffee-script@0.47.2
├── language-csharp@0.12.1
├── language-css@0.37.1
├── language-gfm@0.88.0
├── language-git@0.15.0
├── language-go@0.42.1
├── language-html@0.45.1
├── language-hyperlink@0.16.0
├── language-java@0.23.0
├── language-javascript@0.119.0
├── language-json@0.18.2
├── language-less@0.29.5
├── language-make@0.22.2
├── language-mustache@0.13.0
├── language-objective-c@0.15.1
├── language-perl@0.35.0
├── language-php@0.37.2
├── language-property-list@0.8.0
├── language-python@0.45.0
├── language-ruby@0.69.0
├── language-ruby-on-rails@0.25.0
├── language-sass@0.56.0
├── language-shellscript@0.22.4
├── language-source@0.9.0
├── language-sql@0.23.0
├── language-text@0.7.1
├── language-todo@0.28.0
├── language-toml@0.18.0
├── language-xml@0.34.9
└── language-yaml@0.26.0

Community Packages (16) C:\Users\uc147972\.atom\packages
├── atom-typescript
├── editorconfig@2.0.3
├── file-icons@1.7.24
├── highlight-selected@0.11.2
├── jumpy@3.1.3
├── language-batchfile@0.5.2
├── language-groovy@0.6.0
├── language-typescript-grammars-only@1.5.0
├── linter@1.11.18
├── linter-tslint@0.11.1
├── merge-conflicts@1.4.4
├── minimap@4.25.6
├── pretty-json@1.6.1
├── project-manager@3.3.3
├── tabs-to-spaces@1.0.2
└── todo-show@1.7.0

└── (empty)
```

**Comment by sindresorhus** [comment]
It should be reported as an invalid value in an Atom notification, so you can fix it:

> Invalid value for `max_line_length`. Use `"off"` instead of `0` to disable it.

**Comment by florianb** [comment]
@paulmfischer -- thanks for reporting this. The max_line_length wasn't validated adequate enough. This should be fixed with the v2.0.4-patch. I would appreciate if you could verify it fixes that issue. 💝

**Comment by paulmfischer** [comment]
I don't get an atom notification to fix it even with the new version but it is fixed in v2.0.4 now. Files do not get condensed to one line when the value is 0. Thanks!

**Comment by florianb** [comment]
Thanks @paulmfischer - the notification about improper is slightly more work to do. I opened the separate issue #141 for that.
